src/server/hook_func.c

```diff
@@ -1127,21 +1127,21 @@ py_compile_and_run(char *input_file_path, char *hook_msg, size_t msg_len,
 void
 mgr_hook_import(struct batch_request *preq)
 {
-	svrattrl	*plist, *plx;
-	char		*hookname;
-	hook		*phook;
-	char		content_type[BUFSIZ];
-	char		content_encoding[BUFSIZ];
-	char		input_file[MAXPATHLEN+1];
-	char		input_file_path[MAXPATHLEN+1];
-	char		input_path[MAXPATHLEN+1];
-	char		temp_path[MAXPATHLEN+1];
-	char		output_path[MAXPATHLEN+1];
-	char		hook_msg[HOOK_MSG_SIZE];
-	int		overwrite;
-	struct python_script *py_test_script=NULL;
-	int		rc;
-	int		hook_obj;
+	svrattrl	*plist = NULL, *plx = NULL;
+	char		hookname[PBS_MAXSVRJOBID + 1] = {0};
+	hook		*phook = NULL;
+	char		content_type[BUFSIZ] = {0};
+	char		content_encoding[BUFSIZ] = {0};
+	char		input_file[MAXPATHLEN + 1] = {0};
+	char		input_file_path[MAXPATHLEN + 1] = {0};
+	char		input_path[MAXPATHLEN + 1] = {0};
+	char		temp_path[MAXPATHLEN + 1] = {0};
+	char		output_path[MAXPATHLEN + 1] = {0};
+	char		hook_msg[HOOK_MSG_SIZE] ={0};
+	int		overwrite = -1;
+	struct python_script *py_test_script = NULL;
+	int		rc = -1;
+	int		hook_obj = -1;
```

Comment:
I dislike initializing variables like this for three reasons.  First, if you don't need to do it, you shouldn't.  It takes extra time to zero buffers like this.  Second, while you won't actually read uninitialized memory, you will miss the fact that you're doing something wrong.  It will cover up a bug.  We now build with -Wall -Werror.  If you read from a variable before you initialize it, gcc will throw a warning and travis/jenkins will barf.  Lastly, I've seen codacy complain about this.  You set something just to set it again before using it. 
If you look further down, you memset 3 of your variables you are initializing here to the same thing.  Extra time zeroing buffers for no reason.

Reply:
I think you are right, will remove initialization in next commit.